level.php

```diff
@@ -745,26 +747,55 @@ public function content_tease( $content ) {
 	 */
 	public function user_level_has_cap( $allcaps, $cap, $args ) {
 		if(!$this->_has_global_access() && defined('WPCA_VERSION')) {
-			if(!isset($this->user_levels_caps[$args[1]])) {
-				$this->user_levels_caps[$args[1]] = $allcaps;
-				$levels = $this->get_user_levels($args[1]);
-				if($levels) {
-					//Make sure higher levels have priority
-					//Side-effect: synced levels < normal levels
-					$levels = array_reverse($levels);
-					foreach ($levels as $level) {
-						$level_caps = $this->metadata()->get("caps")->get_data($level);
-						if($level_caps) {
-							foreach ($level_caps as $key => $level_cap) {
-								$this->user_levels_caps[$args[1]][$key] = !!$level_cap;
-							}
-						}
-					}
+			$allcaps = $this->get_user_levels_caps($args[1], $allcaps);
+		}
+		return $allcaps;
+	}
+
+	/**
+	 * Get all user level capabilities (also checks hierarchy)
+	 *
+	 * @since  0.10.x
+	 * @param  int    $user_id
+	 * @param  array  $current_caps (optional preset)
+	 * @return array
+	 */
+	public function get_user_levels_caps($user_id, $current_caps = array()) {
```

Comment:
Get the level caps by user. Used by `user_level_has_cap`
Currently doesn't have a hierarchical parameter it still uses get_user_levels like before.
Could be handy to add this though..